simde/arm/neon/dot_lane.h

```diff
@@ -48,30 +48,36 @@ simde_int32x2_t
 simde_vdot_lane_s32(simde_int32x2_t r, simde_int8x8_t a, simde_int8x8_t b, const int lane)
     SIMDE_REQUIRE_CONSTANT_RANGE(lane, 0, 1) {
   simde_int32x2_t result;
-  #if defined(SIMDE_ARM_NEON_A32V8_NATIVE) && defined(__ARM_FEATURE_DOT_PROD)
-    SIMDE_CONSTIFY_2_(vdot_lane_s32, result, (HEDLEY_UNCREACHABLE(), result), lane, r, a, b);
+  #if defined(SIMDE_ARM_NEON_A64V8_NATIVE) && defined(__ARM_FEATURE_DOTPROD)
+    SIMDE_CONSTIFY_2_(vdot_lane_s32, result, (HEDLEY_UNREACHABLE(), result), lane, r, a, b);
   #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
-    simde_uint32x2_t mask;
-    SIMDE_CONSTIFY_2_(vset_lane_u32, mask, (HEDLEY_UNREACHABLE(), mask), lane, UINT32_MAX, vdup_n_u32(0));
-    result = vbsl_s32(mask,
-                      vadd_s32(r, vmovn_s64(vpaddlq_s32(vpaddlq_s16(vmull_s8(a, b))))),
-                      r);
+    int32_t element;
+    SIMDE_CONSTIFY_2_(vget_lane_s32, element, (HEDLEY_UNREACHABLE(), element), lane,
+                      HEDLEY_STATIC_CAST(simde_int32x2_t, b));
+    simde_int8x8_t b_lane = HEDLEY_STATIC_CAST(simde_int8x8_t, vdup_n_s32(element));
+
+    result = vadd_s32(r, vmovn_s64(
+                         vpaddlq_s32(vpaddlq_s16(
+                         vmull_s8(a, b_lane)))));
```

Comment:
Another minor style issue, but this is kind of ugly.  Here are some other options.

```c
result =
  vadd_s32(
    r,
    vmovn_s64(
      vpaddlq_s32(
        vpaddlq_s16(
          vmull_s8(a, b_lane)
        )
      )
    )
  );
```

```c
result =
  vadd_s32(
    r,
    vmovn_s64(
      vpaddlq_s32(
        vpaddlq_s16(
          vmull_s8(a, b_lane)))));
```

```c
result = vadd_s32(r, vmovn_s64(vpaddlq_s32(vpaddlq_s16(vmull_s8(a, b_lane)))));
```

I'm parial to the first, but any would be okay.  The first two have the advantage of using indentation to help make the code easier to comprehend.  The last one isn't actually that bad; it takes a second to understand, but at least it doesn't have a bunch of random line breaks with no semantic meaning.